core/src/ldml/ldml_processor.cpp

```diff
@@ -198,6 +275,31 @@ ldml_processor::process_event(
         state->context().push_character(str32[i]);
         state->actions().push_character(str32[i]);
       }
+      // add the newly added char
+      ctxt.push_back(str);
+      // Now process transforms
+      std::u16string outputString;
+      // Process the transforms
+      const size_t matchedContext = transforms.matchContext(ctxt, outputString);
+
+      if (matchedContext > 0) {
+        // Found something.
+        // Now, clear out the old context
+        for (size_t i = 0; i < matchedContext; i++) {
+          state->context().pop_back();  // Pop off last
+          auto deletedChar = ctxt[ctxt.size() - i - 1][0];
+          state->actions().push_backspace(KM_KBP_BT_CHAR, deletedChar);  // Cause prior char to be removed
+        }
+        // Now, add in the updated text
+        const std::u32string outstr32 = kmx::u16string_to_u32string(outputString);
+        for (size_t i = 0; i < outstr32.length(); i++) {
+          state->context().push_character(outstr32[i]);
+          state->actions().push_character(outstr32[i]);
+        }
+        // Add a marker so we don't retransform this text.
+        state->context().push_marker(0x0);
+        state->actions().push_marker(0x0);
```

Comment:
Is this written prior to the proposed updates to transforms? As I understand we know no longer only process the remainder.

Reply:
yes, this definitely does not cover monday's proposed updates.